From 8c6111fdbeaa23f6fcdee7963f4d58fd9a79db78 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Tue, 12 Oct 2021 15:45:30 +0530
Subject: [PATCH] Signed-off-by: Amit Prakash Ambasta
 <amit.prakash.ambasta@gmail.com> Use system-catch2 instead of fetchcontent if
 available

---
 CMakeLists.txt              |  1 +
 dependencies/CMakeLists.txt | 22 ++++++++++++++--------
 2 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dd86d2f..542fee4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 include(CTest)
 
 ### Dependencies
+find_package(Catch2 QUIET)
 add_subdirectory(dependencies)
 
 ### Main Build Targets
diff --git a/dependencies/CMakeLists.txt b/dependencies/CMakeLists.txt
index 376ede1..f060a64 100644
--- a/dependencies/CMakeLists.txt
+++ b/dependencies/CMakeLists.txt
@@ -1,13 +1,19 @@
 include(FetchContent)
 
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
-    FetchContent_Declare(
-        catch2
-        GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-    )
+if(
+  CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING AND NOT
+  Catch2_FOUND
+)
+  fetchcontent_declare(
+    catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+  )
 endif()
 
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
-    add_subdirectory(catch2)
+if(
+  CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING AND NOT
+  Catch2_FOUND
+)
+  add_subdirectory(catch2)
 endif()
-add_subdirectory(sqlite3)
\ No newline at end of file
+add_subdirectory(sqlite3)
-- 
2.33.0

